### PR TITLE
[TS-42] resin-release-tool upgrade version to 0.3.0  add github workflow config

### PR DESCRIPTION
**Pull request by thomas-sieren-tmh**
* add github config with workflows for testing and publishing based on the [OCPP repo](https://github.com/mobilityhouse/ocpp)
* add code quality tools  (see the `README.md` changes for the workflow)
* upgrade balena-python-sdk to version with bugfix for https://github.com/balena-io/balena-sdk-python/issues/234
* add basic integration tests
* update poetry lock file

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# https://help.github.com/articles/about-codeowners/
+
+*                         @Roger  @mobilityhouse/qats
```
Comment (thomas-sieren-tmh) [inline .github/CODEOWNERS:3]: @Roger  I created a new qats team recently can you please give write access to the team?  this should then put everybody as a reviewer [automatically](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-teams-and-people-with-access-to-your-repository#inviting-a-team-or-person)

File: .github/workflows/publish-to-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      run: |
+        make install && make test
+
+    - name: Poetry bump version, build and publish
+      shell: bash
+      run: |
+        proj_version=$(poetry version -s)
+        if [ $proj_version != $TAG_VERSION ]; then echo "Version $proj_version, defined in pyproject.toml, does not match TAG $TAG_VERSION of this release"; exit 3; fi
+        poetry update
+        poetry publish --build -u __token__ -p $PYPI_TOKEN
```
Comment (thomas-sieren-tmh) [inline .github/workflows/publish-to-pypi.yml:37]: @OrangeTux , @Roger can you please setup the required secrets for this repo? I think just the PYPI_TOKEN is needed

**Review by OrangeTux — COMMENTED** [review]
~~Can you elaborate on your choice for Github Actions? Our default CI environment is CircleCI. 
We have experience with CircleCI, a lot less with with GH Action.~~

~~We already pay for CircleCI, I'm don't know the cost factor of Github Actions.~~

~~I'm not necessarily against GH Actions, but I prefer a homogeneous set of tools.~~

I just realized that this is an open source project. I think switching to Github Actions makes sense.
GH Actions integrates well with GH and possibly lowers the barrier for contributors to contribute.
Also, we reduce the chances of leaking credentials in our CircleCI environment  accidentally.

File: .github/actions/setup-python-build-env/action.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+name: "Setup Build Environment"
+description: "Install everything needed to build"
+
+runs:
+  using: "composite"
```
Comment (OrangeTux) [inline .github/actions/setup-python-build-env/action.yml:5]: I'm not too familiar with GH Actions. Can you tell me what this line does?

File: .github/actions/setup-python-build-env/action.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  steps:
+  - name: Upgrade pip and install poetry
+    shell: bash
+    run: |
+      pip install --upgrade pip
```
Comment (OrangeTux) [inline .github/actions/setup-python-build-env/action.yml:10]: Can you also pin `pip` to a specific version? I remember some of our builds broke over night when pip changed the way it resolves dependencies.

File: .github/workflows/publish-to-pypi.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+name: Publish 📦 to PyPi 🚀
+
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+*"
```
Comment (OrangeTux) [inline .github/workflows/publish-to-pypi.yml:6]: Usually, our version start with are prefixed with a 'v', like 'v0.9.2'. Can you include the 'v' here?

File: .github/workflows/publish-to-pypi.yml (hunk 2 above)
Comment (OrangeTux) [reply .github/workflows/publish-to-pypi.yml:37]: I don't have access to the project on Pypi.

**Review by OrangeTux — CHANGES_REQUESTED** [review]

**Comment by thomas-sieren-tmh** [comment]
> ~Can you elaborate on your choice for Github Actions? Our default CI environment is CircleCI. We have experience with CircleCI, a lot less with with GH Action.~
> 
> ~We already pay for CircleCI, I'm don't know the cost factor of Github Actions.~
> 
> ~I'm not necessarily against GH Actions, but I prefer a homogeneous set of tools.~
> 
> I just realized that this is an open source project. I think switching to Github Actions makes sense. GH Actions integrates well with GH and possibly lowers the barrier for contributors to contribute. Also, we reduce the chances of leaking credentials in our CircleCI environment accidentally.

Main reason was that the `occp` config is using github. But I also think that having a separation here makes sense. I was also wondering whether there is a risk of accidentally publishing private code to pypi when using `poetry publish` and you forget to add the `-r tmh`  for our private repo. This might just "work" in circle ci if there is a pypi token already configured in a context.

File: .github/actions/setup-python-build-env/action.yml (hunk 3 above)
Comment (thomas-sieren-tmh) [reply .github/actions/setup-python-build-env/action.yml:5]: I'm not sure if it is strictly needed as there is only one run command in this action. 
In general I think it is need combine multiple run statements into a single action 
https://wallis.dev/blog/composite-github-actions

File: .github/workflows/publish-to-pypi.yml (hunk 5 above)
Comment (thomas-sieren-tmh) [reply .github/workflows/publish-to-pypi.yml]: ah, no, sorry I was wrong about the versioning, the `.` is not a wildcard character here

**Review by FireyFly — APPROVED** [review]
LGTM

**Review by Roger — APPROVED** [review]
LGTM

**Pull request merged**